Automattic/jetpack
# Fix double-quotes handling in subscription button

Fixes https://github.com/Automattic/wp-calypso/issues/63196

#### Changes proposed in this Pull Request:

* it adds HTML encoding for double-quote characters used in Subscribe block's submit button

#### Jetpack product discussion



N/A

#### Does this pull request change what data or activity we track or use?



No

#### Testing instructions:






##### Test the fix

1. Open a New Post editor
2. Add Subscribe block
4. Use one or more double-quotes in the button
6. Save the post
7. Open post in the frontend

All double-quotes should be displayed.

Editor view:

![Screen Shot 2022-06-16 at 08 43 39](https://user-images.githubusercontent.com/727413/174008430-829f84d1-19ec-4b36-8753-6f731bf3fa08.png)

Frontend:

![Screen Shot 2022-06-16 at 08 43 32](https://user-images.githubusercontent.com/727413/174008633-53bfa9b6-643b-480b-ad2e-62a32546665b.png)

##### Regression test

We should ensure that posts that were created before the fix can still be edited, and that re-saving such post fixes double-quotes. For that, before the fix is applied, prepare a post that includes Subscribe block, with a button that contains double-quotes.  Then:

1. Edit a Post that includes such a Subscribe block
2. Confirm that block is not broken and there is no prompt for recovery
3. Make any change to the post (add another block, or add a tag), save
4. Confirm that double quotes are displayed in frontend now
